lib/routes/collection.js

```diff
@@ -119,7 +115,10 @@ var routes = function (config) {
       limit:  limit,
       skip:   skip,
     };
+    return query_options;
+  };
 
+  exp._getQuery = function (req, res) {
```

Comment:
I like having these new functions, but the error case when parsing the query in this function might cause issues. Before, the `return` in the case of a bad query would stop execution of the viewCollection() function, but now it'll just stop execution of this new function, and `query` will be some nonsense value on line 147. The session will redirect, but the query will still run (maybe) and who know what happens after that.

Might have to just let the error propagate up, surround line 147 with a try/catch, and then handle the error and redirect there.

(That being said, though it's almost certainly out of scope here, I'd love for the app to restore the query/projection values when there's an error parsing the query.)

Reply:
@MikeyBurkman - good catch!

I moved the try catch blocks up a level so the parent functions are returned on error.